test/bats/azure.bats

```diff
@@ -17,24 +17,101 @@ if [ $TEST_WINDOWS ]; then
   NODE_SELECTOR_OS=windows
 fi
 
-export KEYVAULT_NAME=${KEYVAULT_NAME:-secrets-store-csi-e2e}
+# key-vault configuration
+export KEYVAULT_NAME=${KEYVAULT_NAME:-sscsi-e2e-vault-$(openssl rand -hex 2)}
 export SECRET_NAME=${KEYVAULT_SECRET_NAME:-secret1}
 export SECRET_VERSION=${KEYVAULT_SECRET_VERSION:-""}
 export SECRET_VALUE=${KEYVAULT_SECRET_VALUE:-"test"}
+
+export RESOURCE_GROUP=$(oc get infrastructure cluster -o=jsonpath='{.status.platformStatus.azure.resourceGroupName}')
+export LOCATION=$(az group show --name $RESOURCE_GROUP --query location --output tsv)
+
+
+export OIDC_PROVIDER=$(oc get authentication.config.openshift.io cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+export AZURE_TENANT_ID=$(az account get-access-token --query tenant --output tsv)
+echo $AZURE_TENANT_ID
+
 export LABEL_VALUE=${LABEL_VALUE:-"test"}
 export NODE_SELECTOR_OS=$NODE_SELECTOR_OS
 
 # export the secrets-store API version to be used
 # TODO (aramase) remove this once the upgrade tests are moved to use e2e-provider
 export API_VERSION=$(get_secrets_store_api_version)
 
+
+setup_file(){
+  export USER_ASSIGNED_IDENTITY_NAME="sscsi-e2e-uami-$(openssl rand -hex 2)"
+  echo $USER_ASSIGNED_IDENTITY_NAME
+
+  echo "Create Key Vault"
+  az keyvault create --resource-group "${RESOURCE_GROUP}" \
+   --location "${LOCATION}" \
+   --name "${KEYVAULT_NAME}" \
+   --enable-rbac-authorization false
+
+  echo "Add secret in Key Vault"
+  az keyvault secret set --vault-name "${KEYVAULT_NAME}" \
+   --name "${SECRET_NAME}" \
+   --value "${SECRET_VALUE}"
+
+  echo "Create a user-assigned managed identity"
+  az identity create --name "${USER_ASSIGNED_IDENTITY_NAME}" --resource-group "${RESOURCE_GROUP}"
+
+  echo "Set access policy for the user-assigned managed identity to access the keyvault secret"
+  export IDENTITY_CLIENT_ID="$(az identity show --name "${USER_ASSIGNED_IDENTITY_NAME}" --resource-group "${RESOURCE_GROUP}" --query 'clientId' -otsv)"
+  kubectl annotate serviceaccount default azure.workload.identity/client-id="$IDENTITY_CLIENT_ID" -n default --overwrite
```

Comment:
Why is this annotation required again? As you mentioned(https://github.com/openshift/secrets-store-csi-driver/pull/41#discussion_r2189196323) that it can be removed.